source/simulator/solver_schemes.cc

```diff
@@ -1452,6 +1455,149 @@ namespace aspect
     nonlinear_solver_control.check(1,0.0);
     signals.post_nonlinear_solver(nonlinear_solver_control);
   }
+
+
+
+  template <int dim>
+  void Simulator<dim>::solve_stokes_adjoint ()
+  {
+    // TODO: set an assert here or elsewhere that the material model additive is needed if the max_nonlinear_iteration is more than 1
+    // What is the best way of getting the material model name?
+
+    // This calculation requires that the DT postprocessor is selected
+    AssertThrow(postprocess_manager.template has_matching_postprocessor<const Postprocess::DynamicTopography<dim>> (),
+                ExcMessage("The dynamic topography postprocessor has to be active for the current adjoint computations."));
+
+    // This calculation only works for instantaneous calculations
+    // Should there be an assert that checks for that? I don't see one in the other 'no advection' solvers.
```

Comment:
always. feel free to add one to other solvers :-)

Reply:
Yes. Things that are not supported should just result in an `abort()`, which is what a failed assertion triggers.